src/main.cpp

```diff
@@ -5238,12 +5262,24 @@ bool static ProcessMessage(CNode* pfrom, string strCommand, CDataStream& vRecv,
     {
         vector<CInv> vInv;
         vRecv >> vInv;
-        if (vInv.size() > MAX_INV_SZ)
+        // BU check size == 0 to be intolerant of an empty and useless request
+        if ((vInv.size() > MAX_INV_SZ)||(vInv.size() == 0))
         {
             Misbehaving(pfrom->GetId(), 20);
             return error("message inv size() = %u", vInv.size());
         }
-
+        
+        for (unsigned int nInv = 0; nInv < vInv.size(); nInv++)  // Validate that INVs are a valid type
```

Comment:
the comment is on the same line "// Validate that INVs are a valid type".  Perhaps it needs to be put on the line above and explained more... but that would be yet another commit...